app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -434,14 +440,60 @@ private void setUpGravatar(Message message, MessageHolder messageHolder) {
             float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP,
                     35, resources.getDisplayMetrics());
             String url = message.getSender().getAvatarURL() + "&s=" + px;
+
+            // square default avatar drawable
+            final GradientDrawable defaultAvatar = (GradientDrawable) ContextCompat.getDrawable(context, R.drawable.default_avatar);
```

Comment:
Probably we can move this defaultAvatar from here to onError method itself and pass the old `android.R.drawable.presence_online` in .error() and if error occurs (which should be rare cases for the prod builds) then create this blank drawable!